client.go

```diff
@@ -553,6 +553,12 @@ func (c *Client) Subscribe(params *SubscriptionParameters) (*Subscription, error
 		c,
 	}
 	c.subMux.Lock()
+	if sub.SubscriptionID == 0 || c.subscriptions[sub.SubscriptionID] != nil {
```

Comment:
We could check the subscription id right after the `StatusOK` check and shouldn't you return `nil` instead of `sub` on error?

Reply:
Yeah, I thought about returning `nil`, but _technically_ subscription resources _have_ been created on the server. But realistically, it would be hard to cleanup because I think most operations use the ID. I usually think of `nil` as indicating that nothing was created.

Reply:
The clean-up will never be called b/c the caller checks the error and moves on unless we specifically document that behavior. Then there is nothing useful the caller can do with this. If there is useful cleanup you can do in the client then you should do it here. However, the whole point is that this checks for bugs and edge cases in the server. Lets just return `nil` here unless I'm missing smth substantial.